aws/resource_aws_backup_global_settings.go

```diff
@@ -0,0 +1,59 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/backup"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsBackupGlobalSettings() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsBackupGlobalSettingsUpdate,
+		Update: resourceAwsBackupGlobalSettingsUpdate,
+		Read:   resourceAwsBackupGlobalSettingsRead,
+		Delete: schema.Noop,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"global_settings": {
+				Type:     schema.TypeMap,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceAwsBackupGlobalSettingsUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).backupconn
+
+	input := &backup.UpdateGlobalSettingsInput{
+		GlobalSettings: stringMapToPointers(d.Get("global_settings").(map[string]interface{})),
+	}
+
+	_, err := conn.UpdateGlobalSettings(input)
+	if err != nil {
+		return fmt.Errorf("error setting Backup Global Settings (%s): %w", d.Id(), err)
```

Comment:
When the resource is first created `d.Id()` will not be set yet.
Maybe move `d.SetId()` to earlier in the function.

Reply:
maybe like this?

Reply:
That works.